app/services/customers/generate_checkout_url_service.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Customers
+  class GenerateCheckoutUrlService < BaseService
+    def initialize(customer:)
+      @customer = customer
+      @provider_customer = customer&.provider_customer
+
+      super
+    end
+
+    def call
+      return result.not_found_failure!(resource: 'customer') if customer.blank?
+      return result.service_failure!(
```

Comment:
Style/MultilineIfModifier: Favor a normal if-statement over a modifier clause in a multiline statement. (https://rubystyle.guide#no-multiline-if-modifiers)